src/zimscraperlib/zim/creator.py

```diff
@@ -146,7 +150,55 @@ def config_indexing(
         self.__indexing_configured = True
         return self
 
+    def _log_metadata(self):
+        """Log (DEBUG) all metadata set on (_metadata ~ config_metadata())
+
+        Does not log metadata set post-start (via add_metadata())"""
+        for name, value in sorted(self._metadata.items()):
+            # illustration mandates an Image
+            if re.match(r"^Illustration_(\d+)x(\d+)@(\d+)$", name):
+                try:
+                    with PIL.Image.open(io.BytesIO(value)) as img:
+                        logger.debug(
+                            f"Metadata: {name} is a {len(value)} bytes "
+                            f"{img.size[0]}x{img.size[1]}px {img.format} Image"
+                        )
+                except Exception:
+                    logger.debug(
+                        f"Metadata: {name} is a {len(value)} bytes "
+                        f"{get_content_mimetype(value[:64])} blob "
+                        "not recognized as an Image"
+                    )
+                continue
+
+            # bytes are either encoded string or arbitrary data
+            if isinstance(value, bytes):
+                mimetype = get_content_mimetype(value[:64])
+                if not mimetype.startswith("text/"):
+                    logger.debug(
+                        f"Metadata: {name} is a {len(value)} bytes {mimetype} blob"
+                    )
+                    continue
+                try:
+                    logger.debug(f"Metadata: {name} = {value.decode('UTF-8')}")
+                except Exception:
+                    logger.debug(
+                        f"Metadata: {name} is a {len(value)} bytes {mimetype} blob"
```

Comment:
Should we add ` not decodable as a string with UTF-8 charset`, e.g. `Metadata: Chars-32 is a 16 bytes text/plain blob not decodable as a string with UTF-8 charset`. WDYT? Most metadata are expected to be string, isn't it? I feel like it would make debugging way more straightforward in most cases (and it is harmless for the rare cases where metadata is really a blob).

Reply:
Indeed ; I actually considered doing just that. Fixed and rebased